core/modules/memman/MemManReal.h

```diff
@@ -0,0 +1,77 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_QSERV_MEMMAN_MEMMANREAL_H
+#define LSST_QSERV_MEMMAN_MEMMANREAL_H
+
+// System headers
+#include <atomic>
+#include <mutex>
+#include <unordered_map>
```

Comment:
Granted, that was a left over before a refactored. As for <vector> and <string> those come from MemMan.h that must be included because the class inherits from it. I don't like sprinkling unnecessary includes across various files.